### DISCUSSION TITLE
how to install dubbo from source

### DISCUSSION THREAD

**Discussion by zzps** [apache/dubbo#14962] [Question]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

in actual,I get a bug and it is emergency ,can not wait for next version publish,I want to down load the source and add a fix before next version

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by oxsean** [comment]
After you've configured your maven private repo, replace the dubbo version in pom.xml, then just run:
```bash
./mvnw clean source:jar deploy -T 2C -Dmaven.test.skip=true -Dcheckstyle.skip=true -Dcheckstyle_unix.skip=true -Dmaven.javadoc.skip=true
```

See: https://github.com/apache/dubbo/blob/3.3/.github/workflows/release-test.yml

**Reply by zzps** [reply]
where to config my private repo? i get a  error image 2 when use image 1 config:
image 1:
![1733358632556](https://github.com/user-attachments/assets/012ae7ab-b067-45a6-9594-e3b6028c44d6)
image 2:
![1733358684368](https://github.com/user-attachments/assets/84b792d3-d8c2-4508-b6d9-2b978500df8c)

**Reply by oxsean** [reply]
Your settings.xml

**Reply by zzps** [reply]
> Your settings.xml

i have config settings.xml, 
`<profile>
      <id>nexus2</id>
      <repositories>
        <repository>
          <id>nexus</id>
          <name>Nexus Snapshot Repository</name>
          <url>http://mvnnexus.solarqt.com:8081/repository/maven-public</url>
          <releases>
			<enable>true</enable>
		  </releases>
          <snapshots>
			<enable>true</enable>
		  </snapshots>
        </repository>
      </repositories>
    </profile>`

and i use the cmd
`mvn clean source:jar deploy -T 2C -Dmaven.test.skip=true -Dcheckstyle.skip=true -Dcheckstyle_unix.skip=true -Dmaven.javadoc.skip=true -P nexus2`   
and  it still use 'https://repository.apache.org/content/repositories/snapshots' repo

**Reply by zzps** [reply]
> Your settings.xml

i have config settings.xml, 
`<profile>
      <id>nexus2</id>
      <repositories>
        <repository>
          <id>nexus</id>
          <name>Nexus Snapshot Repository</name>
          <url>http://mvnnexus.solarqt.com:8081/repository/maven-public</url>
          <releases>
			<enable>true</enable>
		  </releases>
          <snapshots>
			<enable>true</enable>
		  </snapshots>
        </repository>
      </repositories>
    </profile>`

and i use the cmd
`mvn clean source:jar deploy -T 2C -Dmaven.test.skip=true -Dcheckstyle.skip=true -Dcheckstyle_unix.skip=true -Dmaven.javadoc.skip=true -P nexus2`   
and  it still use 'https://repository.apache.org/content/repositories/snapshots' repo

**Reply by oxsean** [reply]
Please ask Google or chatgpt about detail, In my memory your should modify config distributionManagement in dubbo's pom.xml, and add server into settings.xml.

**Reply by zzps** [reply]
OK  . I fix it
